README.md

```diff
@@ -32,6 +32,8 @@ General guidelines for Hellbender developers
 
 * Untested code is considered **non-existent** and thus is subject to removal at any time (exception is main methods, or super corner conditions, or `toString()` code).
 
+* Each tool should have at least one good end-to-end integration test with a check for expected output, plus high-quality unit tests for all non-trivial utility methods/classes used by the tool. Although we have no specific coverage target, coverage should be extensive enough that if tests pass, the tool is guaranteed to be in a usable state.
```

Comment:
How about "the tool is guaranteed to be in a usable state for its intended use(s)"? :)

For new tools, defining test cases is typically an iterative process: the initial set of test cases is proposed by the tool author in the initial pull request, then this set is potentially modified during code review if the reviewer notices significant cases that are missing. Additional test cases will then potentially be added post-merge as bug reports come in from the users.

For existing tools that lack tests, the process is much the same, except that the original tool author may not be available to write the initial set of test cases, so the task is more difficult. If the tool has an existing userbase, those who interact with the users (eg., the support group) could help devise a set of important use cases that should be covered by tests. If the tool has no userbase, we should probably debate whether it should remain in the codebase at all.

@akiezun @lbergelson please feel free to weigh in here as well